### ISSUE TITLE
visual representation of selected dates wrong after select and instant deselect

### ISSUE THREAD

**Issue by christian-steinmeyer** [spongessuck/gm.datepickerMultiSelect#21]
I'm using the multi-select within a modal (but think, that shouldn't make a difference). When selecting multiple dates, say August 12th, 14th and 15th (in this order) 
![multi-select-correct](https://cloud.githubusercontent.com/assets/7511021/9245690/72034382-41a6-11e5-94f5-b6e8b7dc504c.jpg)

and deselecting the most recently selected date again (clicking on August 15th again), within the inline calendar, all elements are displayed as if not selected, even though the model stays correct and only deselects August 15th as desired. 
![multi-select-false-after-deselect](https://cloud.githubusercontent.com/assets/7511021/9245693/72066a12-41a6-11e5-9dfb-b6dfc9664e35.jpg)

Selecting August 15th again results in all elements being displayed correctly again, but Selecting a different date does not trigger the correct display again until August 15th is selected again (although this triggered the false display of the dates selected in the mean time again).
![multi-select-false-after-deselect-plus-new-select](https://cloud.githubusercontent.com/assets/7511021/9245694/7206cfd4-41a6-11e5-95ba-a7b6017a3285.jpg)
![multi-select-false-after-deselect-plus-new-select-plus-select-15-again](https://cloud.githubusercontent.com/assets/7511021/9245692/7205f4f6-41a6-11e5-9480-27d1b4d3789a.jpg)
![multi-select-false-after-deselect-plus-new-select-plus-select-and-deselect-15-again](https://cloud.githubusercontent.com/assets/7511021/9245691/7204827e-41a6-11e5-9dc0-054e40ee2fd0.jpg)

Here is my markup
```html
<datepicker ng-model='model.activeDate' multi-select='model.dates'></datepicker>
<p ng-repeat="date in model.dates">{{date | date:"fullDate"}}</p>
```

and my Javascript
```javascript
$scope.model.activeDate;
$scope.model.dates = [];
```

and some of the dependencies of bower
```
{
  ...
  "dependencies": {
    "angular": "^1.3.0",
    "bootstrap": "3.3.4",
   ...
    "angular-bootstrap": "~0.13.2",
    "gm.datepicker-multi-select": "~1.0.6"
  },
 ...
}
```

**Comment by spongessuck** [comment]
I tried it a few different ways and this one works fine:

http://plnkr.co/edit/foSW3YzZVCHJX11PZTM9?p=preview

For some reason when the model is in the parent scope it didn't list the dates, but the dates highlighted properly on the calendar. I suspect it's a scope issue.

**Comment by christian-steinmeyer** [comment]
Thank you for the quick response! I applied your changes to my code, but discovered, that it was still not working. Then I applied my code to your example and that worked (http://plnkr.co/edit/zP0Sk3E8g5TgxRoTE6V7?p=preview). So I assume the error must be caused by some difference in project setup. I'm not sure what it is, though. I am using the exact same versions of angular and bootstrap. I used yeoman with bower and grunt to create a quick app, but I don't think that should really cause this trouble. Thank you for your help, though!
